### PR TITLE
Adding differential tests

### PR DESCRIPTION
**Pull request by kopy-kat**

### CODE REVIEW

File: test/differential/Biconomy.t.sol [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        instance.setCondition(newExecutor, conditions);
+
+        bytes32 digestOnManager =
+            instance.aux.compConditionManager.getHash(instance.account, newExecutor);
+        assertEq(digest, digestOnManager);
+    }
+
+    function testAddFallback() public {
+        TokenReceiver handler = new TokenReceiver();
+        string memory functionSig = "onERC721Received(address,address,uint256,bytes)";
```
Comment (zeroknots) [inline test/differential/Biconomy.t.sol:213]: should we use IERC721.onERC721Received.selector here for better readability?

File: test/mocks/fallback/TokenReceiver.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        uint256 value,
+        bytes calldata data
+    )
+        external
+        view
+        override
+        returns (bytes memory result)
+    {
+        console2.log("Handling fallback");
+        bytes4 selector = 0x150b7a02;
```
Comment (zeroknots) [inline test/mocks/fallback/TokenReceiver.sol:17]: which selector is this?

**Comment by zeroknots** [comment]
@kopy-kat I resolved the review comments in the commits. please review especially https://github.com/rhinestonewtf/modulekit/pull/55/commits/1e5c8ea7b645fc175b8a2ff568591fb3214eca18

**Comment by zeroknots** [comment]
@kopy-kat lets hold off with the merge to main for a bit. Im rebasing feature/kernel-integration branch ontop of feature/test-scaffold and will PR into here. will make merging a bit easier

File: test/mocks/fallback/TokenReceiver.sol (hunk 2 above)
Comment (kopy-kat) [reply test/mocks/fallback/TokenReceiver.sol]: IERC721.onERC721Received.selector

**Pull request merged**